remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/RemoteMessagingConfigDownloadScheduler.kt

```diff
@@ -56,14 +61,31 @@ class RemoteMessagingConfigDownloadWorker(
     scope = AppScope::class,
     boundType = MainProcessLifecycleObserver::class,
 )
+
+@ContributesMultibinding(
+    AppScope::class,
+    boundType = PrivacyConfigCallbackPlugin::class,
+)
 class RemoteMessagingConfigDownloadScheduler @Inject constructor(
     private val workManager: WorkManager,
-) : MainProcessLifecycleObserver {
+    private val downloader: RemoteMessagingConfigDownloader,
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+    private val remoteMessagingConfigRepository: RemoteMessagingConfigRepository,
+) : MainProcessLifecycleObserver, PrivacyConfigCallbackPlugin {
 
     override fun onCreate(owner: LifecycleOwner) {
         scheduleDownload()
     }
 
+    override fun onPrivacyConfigDownloaded() {
```

Comment:
Can we add a quick killswitch around this logic in case something unexpected goes wrong? so just create a RMFFeature with a subfeature `invalidateAfterPrivacyConfigDownloaded`, default true. Just safety net if this creates any unexpected side effect.

Reply:
I added the kill switch and unit tests around it.

Reply:
Thanks!